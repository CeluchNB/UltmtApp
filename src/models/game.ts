import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { Tournament } from '../types/tournament'
import { CreateGame, Game, GameStatus, PointStats } from '../types/game'
import { DisplayTeam, GuestTeam } from '../types/team'

export class GameSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Game',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            creator: 'DisplayUser',
            teamOne: 'DisplayTeam',
            teamTwo: 'GuestTeam',
            teamTwoDefined: 'bool',
            scoreLimit: 'int',
            halfScore: 'int',
            startTime: 'date',
            softcapMins: 'int',
            hardcapMins: 'int',
            playersPerPoint: 'int',
            timeoutPerHalf: 'int',
            floaterTimeout: 'bool',
            tournament: 'Tournament?',
            teamOneScore: 'int',
            teamTwoScore: 'int',
            teamOnePlayers: 'DisplayUser[]',
            teamTwoPlayers: 'DisplayUser[]',
            resolveCode: 'string',
            statsPoints: 'PointStats[]',
            offline: 'bool',
            teamOneStatus: 'string',
            teamTwoStatus: 'string',
        },
    }

    _id: string
    creator: DisplayUser
    teamOne: DisplayTeam
    teamTwo: GuestTeam
    teamTwoDefined: boolean
    scoreLimit: number
    halfScore: number
    startTime: Date
    softcapMins: number
    hardcapMins: number
    playersPerPoint: number
    timeoutPerHalf: number
    floaterTimeout: boolean
    tournament?: Tournament
    teamOneScore: number
    teamTwoScore: number
    teamOnePlayers: DisplayUser[]
    teamTwoPlayers: DisplayUser[]
    resolveCode: string
    statsPoints: PointStats[]
    offline: boolean
    teamOneStatus: GameStatus
    teamTwoStatus: GameStatus

    constructor(
        game: Game,
        offline: boolean = false,
        statsPoints: PointStats[] = [],
    ) {
        this._id = game._id
        this.creator = game.creator
        this.teamOne = game.teamOne
        this.teamTwo = game.teamTwo
        this.teamTwoDefined = game.teamTwoDefined
        this.scoreLimit = game.scoreLimit
        this.halfScore = game.halfScore
        this.startTime = game.startTime
        this.softcapMins = game.softcapMins
        this.hardcapMins = game.hardcapMins
        this.playersPerPoint = game.playersPerPoint
        this.timeoutPerHalf = game.timeoutPerHalf
        this.floaterTimeout = game.floaterTimeout
        this.tournament = game.tournament
        this.teamOneScore = game.teamOneScore
        this.teamTwoScore = game.teamTwoScore
        this.teamOnePlayers = game.teamOnePlayers
        this.teamTwoPlayers = game.teamTwoPlayers
        this.resolveCode = game.resolveCode || ''
        this.statsPoints = statsPoints
        this.offline = offline
        this.teamOneStatus = game.teamOneStatus
        this.teamTwoStatus = game.teamTwoStatus
    }

    static createOfflineGame(game: CreateGame, teamOnePlayers: DisplayUser[]) {
        return {
            _id: new Realm.BSON.ObjectID().toHexString(),
            creator: game.creator,
            teamOne: game.teamOne,
            teamTwo: game.teamTwo,
            teamTwoDefined: game.teamTwoDefined,
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
            tournament: game.tournament,
            teamOneScore: 0,
            teamTwoScore: 0,
            teamOneActive: true,
            teamTwoActive: false,
            teamOnePlayers: teamOnePlayers,
            teamTwoPlayers: [],
            resolveCode: '',
            statsPoints: [],
            offline: true,
        }
    }
}
export const PointStatsSchema: Realm.ObjectSchema = {
    name: 'PointStats',
    embedded: true,
    properties: {
        _id: 'string',
        pointStats: 'PlayerPointStats[]',
    },
}

export const PlayerPointStatsSchema: Realm.ObjectSchema = {
    name: 'PlayerPointStats',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string?',
        pointsPlayed: 'int',
        goals: 'int',
        assists: 'int',
        turnovers: 'int',
        blocks: 'int',
    },
}
