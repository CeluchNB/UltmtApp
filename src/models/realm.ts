import { Realm } from '@realm/react'
import {
    ActionSchema,
    CommentSchema,
    DisplayTeamSchema,
    DisplayUserSchema,
    GameSchema,
    GuestTeamSchema,
    PointSchema,
    TournamentSchema,
} from './index'

const SCHEMAS = [
    DisplayTeamSchema,
    DisplayUserSchema,
    GameSchema.schema,
    GuestTeamSchema,
    TournamentSchema,
    PointSchema.schema,
    ActionSchema.schema,
    CommentSchema,
]

const config = {
    schema: SCHEMAS,
    shouldCompactOnLaunch: function (_totalSize: number, _usedSpace: number) {
        // TODO: should implement this, but never seems to be called?
        return false
    },
}

let realm: Realm | undefined

export const getRealm = async (): Promise<Realm> => {
    try {
        if (!realm || realm.isClosed) {
            realm = await Realm.open(config)
        }
        return realm
    } catch (error) {
        throw error
    }
}

export const closeRealm = () => {
    realm?.close()
}
