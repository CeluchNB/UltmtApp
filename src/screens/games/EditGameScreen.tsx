import * as Constants from '../../utils/constants'
import BaseScreen from '../../components/atoms/BaseScreen'
import { EditGameProps } from '../../types/navigation'
import GameForm from '../../components/organisms/GameForm'
import PrimaryButton from '../../components/atoms/PrimaryButton'
import React from 'react'
import { useForm } from 'react-hook-form'
import { ScrollView, StyleSheet, Text } from 'react-native'
import { useGameEditor, useTheme } from '../../hooks'

const EditGameScreen: React.FC<EditGameProps> = ({ navigation }) => {
    const { game, onEditGame } = useGameEditor()
    const {
        theme: { colors, size },
    } = useTheme()
    const [loading, setLoading] = React.useState(false)
    const [error, setError] = React.useState('')

    const {
        control,
        handleSubmit,
        formState: { errors },
    } = useForm({
        defaultValues: {
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
        },
    })

    const onSubmit = async () => {
        try {
            setLoading(true)
            await handleSubmit(onEditGame)()
            navigation.goBack()
        } catch (e: any) {
            setError(e?.message ?? Constants.UPDATE_GAME_ERROR)
        } finally {
            setLoading(false)
        }
    }

    const styles = StyleSheet.create({
        errorText: {
            color: colors.error,
            fontSize: size.fontTen,
        },
    })
    return (
        <ScrollView>
            <BaseScreen containerWidth={80}>
                <GameForm control={control} errors={errors} />
                {error.length > 0 && (
                    <Text style={styles.errorText}>{error}</Text>
                )}
                <PrimaryButton
                    text="make updates"
                    loading={loading}
                    onPress={onSubmit}
                />
            </BaseScreen>
        </ScrollView>
    )
}

export default EditGameScreen
