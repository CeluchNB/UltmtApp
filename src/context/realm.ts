import { config } from '../models/realm'
import { createRealmContext } from '@realm/react'

const context = createRealmContext(config)

export const { RealmProvider, useObject } = context
