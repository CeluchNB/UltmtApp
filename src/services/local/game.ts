import * as Constants from '../../utils/constants'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { DisplayUser } from '../../types/user'
import { getRealm } from '../../models/realm'
import { throwApiError } from '../../utils/service-utils'
import { ActionSchema, GameSchema, PointSchema } from '../../models'
import { CreateGame, Game } from '../../types/game'

const parseGame = (schema: GameSchema): Game & { offline: boolean } => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id,
            creator: schema.creator,
            teamOne: schema.teamOne,
            teamTwo: schema.teamTwo,
            teamTwoDefined: schema.teamTwoDefined,
            scoreLimit: schema.scoreLimit,
            halfScore: schema.halfScore,
            startTime: schema.startTime,
            softcapMins: schema.softcapMins,
            hardcapMins: schema.hardcapMins,
            playersPerPoint: schema.playersPerPoint,
            timeoutPerHalf: schema.timeoutPerHalf,
            floaterTimeout: schema.floaterTimeout,
            tournament: schema.tournament,
            teamOneScore: schema.teamOneScore,
            teamTwoScore: schema.teamTwoScore,
            teamOneActive: schema.teamOneActive,
            teamTwoActive: schema.teamTwoActive,
            teamOnePlayers: schema.teamOnePlayers,
            teamTwoPlayers: schema.teamTwoPlayers,
            resolveCode: schema.resolveCode,
            points: schema.points,
            offline: schema.offline,
        }),
    )
}

/**
 * Method to determine if currently active game is offline
 * @returns boolean indicating current game is offline or not
 */
export const activeGameOffline = async (): Promise<boolean> => {
    return (await AsyncStorage.getItem('active_game_offline')) === 'true'
}

/**
 * Method to get the ObjectId of the current active game
 * @returns current active game id
 */
export const activeGameId = async (): Promise<string> => {
    return (await AsyncStorage.getItem('active_game_id')) || ''
}

/**
 * Method to set active game offline status
 * @param offline boolean
 */
export const setActiveGameOffline = async (offline: boolean) => {
    await AsyncStorage.setItem('active_game_offline', offline.toString())
}

/**
 * Method to set active game id
 * @param id string
 */
export const setActiveGameId = async (id: string) => {
    await AsyncStorage.setItem('active_game_id', id)
}

export const createOfflineGame = async (
    data: CreateGame,
    teamOnePlayers: DisplayUser[],
): Promise<string> => {
    const realm = await getRealm()

    let id: string = ''
    realm.write(() => {
        const game = realm.create<GameSchema>(
            'Game',
            GameSchema.createOfflineGame(data, teamOnePlayers),
        )
        id = game._id
    })

    return id
}

export const saveGame = async (game: Game) => {
    const realm = await getRealm()
    const gameRecord = await realm.objectForPrimaryKey<GameSchema>(
        'Game',
        game._id,
    )
    let offline = false
    if (gameRecord) {
        offline = gameRecord.offline
    }
    const schema = new GameSchema(game, offline)
    realm.write(() => {
        realm.create('Game', schema, Realm.UpdateMode.Modified)
    })
}

export const getGameById = async (
    gameId: string,
): Promise<Game & { offline: boolean }> => {
    const realm = await getRealm()
    const game = await realm.objectForPrimaryKey<GameSchema>('Game', gameId)
    if (!game) {
        return throwApiError({}, Constants.GET_GAME_ERROR)
    }
    return parseGame(game)
}

export const activeGames = async (
    userId: string,
): Promise<(Game & { offline: boolean })[]> => {
    const realm = await getRealm()
    const games = await realm.objects<GameSchema>('Game')

    return games
        .filter(g => g.creator._id === userId)
        .map(game => parseGame(game))
}

export const deleteFullGame = async (gameId: string): Promise<void> => {
    const realm = await getRealm()
    const game = await realm.objectForPrimaryKey<GameSchema>('Game', gameId)

    const points = game?.points.map(id => {
        return realm.objectForPrimaryKey<PointSchema>('Point', id)
    })

    const actions: (ActionSchema | null)[] = []
    points?.forEach(p => {
        p?.teamOneActions.forEach(id => {
            const objectId = new Realm.BSON.ObjectID(id)
            actions.push(
                realm.objectForPrimaryKey<ActionSchema>('Action', objectId),
            )
        })
        p?.teamTwoActions.forEach(id => {
            const objectId = new Realm.BSON.ObjectID(id)
            actions.push(
                realm.objectForPrimaryKey<ActionSchema>('Action', objectId),
            )
        })
    })

    realm.write(() => {
        realm.delete(actions.filter(a => a !== null))
        realm.delete(points?.filter(p => p !== null))
        realm.delete(game)
    })
}
