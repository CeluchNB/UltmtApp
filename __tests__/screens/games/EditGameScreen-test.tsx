import * as GameData from '../../../src/services/data/game'
import { EditGameProps } from '../../../src/types/navigation'
import EditGameScreen from '../../../src/screens/games/EditGameScreen'
import { NavigationContainer } from '@react-navigation/native'
import React from 'react'
import { RealmProvider } from '../../../src/context/realm'
import { game } from '../../../fixtures/data'
import { QueryClient, QueryClientProvider } from 'react-query'
import { fireEvent, render, waitFor } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const goBack = jest.fn()
const props: EditGameProps = {
    navigation: {
        goBack,
    } as any,
    route: { params: { gameId: '' } } as any,
}

beforeAll(() => {
    jest.useFakeTimers({ legacyFakeTimers: true })
})

afterAll(() => {
    jest.useRealTimers()
})

const client = new QueryClient()

describe('EditGameScreen', () => {
    it('matches snapshot', () => {
        const snapshot = render(
            <RealmProvider>
                <QueryClientProvider client={client}>
                    <NavigationContainer>
                        <EditGameScreen {...props} />
                    </NavigationContainer>
                </QueryClientProvider>
            </RealmProvider>,
        )

        expect(snapshot.getByText('Game to')).toBeTruthy()
        expect(snapshot.getByText('Half at')).toBeTruthy()
        expect(snapshot.getByText('make updates')).toBeTruthy()
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    // it('handles update press', async () => {
    //     const { getByText } = render(
    //         <NavigationContainer>
    //             <EditGameScreen {...props} />
    //         </NavigationContainer>,
    //     )

    //     const updateBtn = getByText('make updates')
    //     fireEvent.press(updateBtn)

    //     expect(goBack).toHaveBeenCalled()
    // })
})
