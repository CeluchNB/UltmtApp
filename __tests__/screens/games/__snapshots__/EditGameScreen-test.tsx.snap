// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGameScreen matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#ffd600",
            "fontSize": 61.64383561643835,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Edit Game
    </Text>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Game to
        </Text>
        <View
          style={
            {
              "width": "30%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "#303030",
                    "borderColor": "#31adff",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="number-pad"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Game to"
                  placeholderTextColor="#ffffff"
                  selectionColor="#ffd600"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value="15"
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#ababab",
              "fontSize": 30.821917808219176,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          points
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Half at
        </Text>
        <View
          style={
            {
              "width": "30%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "#303030",
                    "borderColor": "#31adff",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="number-pad"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Half at"
                  placeholderTextColor="#ffffff"
                  selectionColor="#ffd600"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value="8"
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#ababab",
              "fontSize": 30.821917808219176,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          points
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Soft cap
        </Text>
        <View
          style={
            {
              "width": "30%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "#303030",
                    "borderColor": "#31adff",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="number-pad"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Soft cap"
                  placeholderTextColor="#ffffff"
                  selectionColor="#ffd600"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value="75"
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#ababab",
              "fontSize": 30.821917808219176,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          minutes
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Hard cap
        </Text>
        <View
          style={
            {
              "width": "30%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "#303030",
                    "borderColor": "#31adff",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="number-pad"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Hard cap"
                  placeholderTextColor="#ffffff"
                  selectionColor="#ffd600"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value="90"
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#ababab",
              "fontSize": 30.821917808219176,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          points
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Players
        </Text>
        <View
          style={
            {
              "width": "30%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "#303030",
                    "borderColor": "#31adff",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="number-pad"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Players"
                  placeholderTextColor="#ffffff"
                  selectionColor="#ffd600"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value="7"
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#ababab",
              "fontSize": 30.821917808219176,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          per point
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Timeouts
        </Text>
        <View
          style={
            {
              "width": "30%",
            }
          }
        >
          <View
            style={{}}
          >
            <View
              pointerEvents="none"
              style={
                [
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  {
                    "backgroundColor": "#303030",
                    "borderColor": "#31adff",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  autoCapitalize="none"
                  editable={true}
                  keyboardType="number-pad"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="Timeouts"
                  placeholderTextColor="#ffffff"
                  selectionColor="#ffd600"
                  style={
                    [
                      {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "#ffd600",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value="1"
                />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#ababab",
              "fontSize": 30.821917808219176,
              "fontWeight": "bold",
              "marginLeft": 10,
            }
          }
        >
          per half
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 5,
            "marginTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffd600",
              "fontSize": 41.0958904109589,
              "fontWeight": "bold",
              "width": "50%",
            }
          }
        >
          Floater timeout
        </Text>
        <View
          style={
            {
              "width": "50%",
            }
          }
        >
          <RCTSwitch
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onTintColor="#31adff"
            style={
              {
                "height": 31,
                "width": 51,
              }
            }
            thumbTintColor="#ffd600"
            tintColor="#ababab"
            value={true}
          />
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "alignSelf": undefined,
          "bottom": undefined,
          "end": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#ffd600",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "minWidth": "auto",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                false,
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      {
                        "marginHorizontal": 8,
                      },
                      {
                        "textTransform": "uppercase",
                      },
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                make updates
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
